docs/source/conf.py

```diff
@@ -158,11 +206,18 @@
 # built documents.
 #
 # The short X.Y version.
-version = os.environ.get('ARROW_DOCS_VERSION',
-                         pyarrow.__version__)
+try:
+    version = os.environ.get('ARROW_DOCS_VERSION',
+                             pyarrow.__version__)
```

Comment:
I think you could try to avoid passing `pyarrow.__version__` to this function, as that will fail if pyarrow wasn't imported, even when ARROW_DOCS_VERSION is specified. 

So that we allow you to set the version with that env variable, also for the case pyarrow isn't installed? (I don't know how important this use case is, though)

Reply:
So yeah, we could default to an empty string if `ARROW_DOCS_VERSION` is not specified. I guess that can't happen (env var not being set) when publishing the docs with the release, right?

Reply:
> So yeah, we could default to an empty string if `ARROW_DOCS_VERSION` is not specified.

I think it would be nice to still use `pyarrow.__version__` _if_ it is installed, and the env variable is not set. It's nice to see a correct version number when building the docs locally for development (when you typically don't specify that env variable).

You could for example use `pyarrow_version =  pyarrow.__version` and `pyarrow_version = ""` in the if/else branches above when trying to import pyarrow, and then use that variable here (`version = os.environ.get('ARROW_DOCS_VERSION', pyarrow_version)`)